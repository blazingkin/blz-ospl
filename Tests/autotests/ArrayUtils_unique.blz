:main
    #Test 00 :unique
    print("Test 00 unique on sorted array")
    arr = [1,2,2,3,3,3,4,4,5,5,6]
    arr2 = arr.unique()
    print(arr2)                                   # unique array (removed duplicates)
    print(arr)                                    # original array (unchanged)
    print("--------------------------------")

    #Test 01 :unique                                       #does not work if the array is unsorted
    print("Test 01 unique on unsorted array")
    arr = [3,4,2,3,9,2,6,0,1,4,8,3,3,7,9]
    arr2 = arr.unique()
    print(arr2)
    print(arr)
    print("--------------------------------")
    
end