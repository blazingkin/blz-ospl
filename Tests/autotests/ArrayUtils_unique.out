Test 00 unique on sorted array
[1, 2, 3, 4, 5, 6]
[1, 2, 2, 3, 3, 3, 4, 4, 5, 5, 6]
--------------------------------
Test 01 unique on unsorted array
[3, 4, 2, 6, 0, 1, 8, 7, 9]
[3, 4, 2, 3, 9, 2, 6, 0, 1, 4, 8, 3, 3, 7, 9]
--------------------------------
